### PR TITLE
Feature/reference data

### PR DESCRIPTION
**Pull request by rivexe**
**referenceData** for PHP, Python, CSharp, CSharp-MVC, Ruby, Java, Java-Spring examples.

### CODE REVIEW

File: web/documentserver-example/php/webeditor-ajax.php [hunk 1]
```diff
@@ -570,3 +574,52 @@ function renamefile()
...
+        if ($instanceId == serverPath()) {
+            $fileKey = json_decode($referenceData["fileKey"]);
+            $userAddress = $fileKey->userAddress;
+            if ($userAddress == getCurUserHostAddress()) {
+                $fileName = $fileKey->fileName;
+            }
+        }
+    }
+
+    if (!$fileName) {
```
Comment (LinneyS) [inline web/documentserver-example/php/webeditor-ajax.php:602]: Add search for `fileName` by `path` like this
https://github.com/ONLYOFFICE/document-server-integration/commit/b27e532a830c2f8a6a3df0225bde487a0ec4a842#diff-7398df16c124c271406c203da02d406e5a6bc55926e9e224b52b0f68d99f4097R497-R503

**Review by LinneyS — CHANGES_REQUESTED** [review]

File: web/documentserver-example/php/webeditor-ajax.php (hunk 1 above)
Comment (rivexe) [reply web/documentserver-example/php/webeditor-ajax.php]: Search for `fileName` by `path` added + some possible exceptions in AJAX handled.

File: web/documentserver-example/php/webeditor-ajax.php [hunk 2]
```diff
@@ -570,3 +574,59 @@ function renamefile()
...
+            if ($userAddress == getCurUserHostAddress()) {
+                $fileName = $fileKey->fileName;
+            }
+        }
+    }
+
+    if (!isset($filename) && isset($post["path"]) && isset($userAddress)) {
+        $path = basename($post["path"]);
+        if (file_exists(getStoragePath($path, $userAddress))) {
+            $fileName = $path;
```
Comment (LinneyS) [inline web/documentserver-example/php/webeditor-ajax.php:605]: does not work

**Review by LinneyS — CHANGES_REQUESTED** [review]

File: web/documentserver-example/php/webeditor-ajax.php (hunk 2 above)
Comment (rivexe) [reply web/documentserver-example/php/webeditor-ajax.php:605]: filxed

File: web/documentserver-example/csharp-mvc/WebEditor.ashx.cs [hunk 3]
```diff
@@ -591,5 +594,96 @@ private static void Rename(HttpContext context)
...
+            if (JwtManager.Enabled)
+            {
+                var token = JwtManager.Encode(data);
+                data.Add("token", token);
+            }
+
+            context.Response.Write(jss.Serialize(data).Replace(@"\u0026", "&"));
+        }
+
+    }
```
Comment (LinneyS) [inline web/documentserver-example/csharp-mvc/WebEditor.ashx.cs:687]: unnecessary

File: web/documentserver-example/csharp-mvc/Models/FileModel.cs [hunk 4]
```diff
@@ -138,6 +138,17 @@ public string GetDocConfig(HttpRequest request, UrlHelper url)
                                             { "favorite", favorite}
                                         }
                                 },
+                                {
+                                    "referenceData", new Dictionary<string, string>()
+                                    {
+                                        { "fileKey", !user.id.Equals("uid-0") ?
+                                            jss.Serialize(new Dictionary<string, object>{
+                                                {"fileName", FileName},
+                                                {"userAddress", HttpUtility.UrlEncode(_Default.CurUserHostAddress(HttpContext.Current.Request.UserHostAddress))}
```
Comment (LinneyS) [inline web/documentserver-example/csharp-mvc/Models/FileModel.cs:147]: `_Default` ?

**Review by LinneyS — CHANGES_REQUESTED** [review]

File: web/documentserver-example/csharp-mvc/WebEditor.ashx.cs (hunk 3 above)
Comment (rivexe) [reply web/documentserver-example/csharp-mvc/WebEditor.ashx.cs:687]: deleted

File: web/documentserver-example/csharp-mvc/Models/FileModel.cs (hunk 4 above)
Comment (rivexe) [reply web/documentserver-example/csharp-mvc/Models/FileModel.cs]: fixed

**Pull request merged**